### ISSUE TITLE
How to keep the visible unit with missing value always off?

### ISSUE THREAD

**Issue by michelleowen** [felipecruz/CFRBM#3]
There are a lot of missing values in my data. I understand that in the training phase, we need to turn those units always off including the reconstruction phase. Apparently only setting them as zeros in the training data is not enough. How can I do this in theano?

**Comment by felipecruz** [comment]
What CFRBM do is to multiply the visible layer vector element-wise with the original "mask". When you go from the hidden layer to the visible layer all of them will have non zero values. When you multiply them element-wise with the original mask you turn to 0 all values that shouldn't be consideres in the update. It's just a element-wise multiplication. Hope it helped

**Comment by michelleowen** [comment]
Brilliant!
A side question, I understand that CRBM can sort of handle cold start problem. However, it seems that one can only solve either new users or new items problem. On Cross-Validation website, I saw you mentioned there is a paper on UI-RBM. Could you share the paper?

**Comment by felipecruz** [comment]
@michelleowen sorry for the delayed response: http://jmlr.org/proceedings/papers/v28/georgiev13.pdf

**Comment by michelleowen** [comment]
Thanks!
